book/src/component-model/tasks.md

```diff
@@ -0,0 +1,36 @@
+## Tasks
+
+A `Task` records that an instance of a process was run in a Harbor environment. It may be helpful to
+think of a task as unit of work. The enrichment engine is implemented as a set of services that 
+implement the `TaskProvider` trait. You can build a service that implements the `TaskProvider` 
+trait and unless you override the key default functions, Harbor will record details about each 
+instantiation of the process by storing an instance of a `Task` entity.
+
+The `TaskProvider` trait has the following interface. 
+
+```rust
+pub trait TaskProvider: Service<Task> {
+    /// Implement this with your custom logic.
+    async fn run(&self, task: &mut Task) -> Result<HashMap<String, String>, Error>;
+
+    /// Runs the transaction script and store the results. Usually invoked by a CLI command handler.
+    async fn execute(&self, task: &mut Task) -> Result<(), Error> {
```

Comment:
1. It is unintuitive to me as to why we have a `run()` method _and_ an `execute()` method.   They seem to serve the same purpose. 
2. I also don't understand the reference to a `transaction script`.  
3. Does the information regarding the CLI come before this?  Should we add links to that area of the documentation to help readers navigate the different concepts we have in Harbor?

Reply:
Hmmmm.  Good callouts. 
1. I hear what you are saying. Got any suggestions for a better name than `run`?
2. This should actually be Template Method. I'll add a link to the explanation in the Trait definition and update this language to be more generic.
3. Did you run the book locally? In theory, it has a flow that prepares you for the subsequent content. I'd like to limit hard links as much as possible because as I learned at `$lastJob`, they become a source of maintenance toil. There isn't a way to "find all links" when making documentation changes or restructuring directories. This often leads to dead links and user frustration.

Reply:
Run might be good, but why do we need two?  Can we survive with _just_ an _execute()_ method?

Reply:
No, I have not run the book locally.  Didn't even know one could run a book...

Reply:
I'll figure out how to run it